OmgvaPOS/src/OrderManagement/Controller/OrderController.cs

```diff
@@ -0,0 +1,135 @@
+            return Forbid();
+
+        try {
+            OrderDTO orderDTO = _orderService.GetOrder(id);
+            return Ok(orderDTO);
+        }
+        catch (Exception ex) {
+            _logger.LogError(ex, "An unexpected internal server error occured while trying to get an order.");
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+        }
+    }
+
+    [HttpGet]
+    [Authorize(Roles = "Admin,Owner,Employee")]
+    [ProducesResponseType<OrderDTO>(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public ActionResult<IEnumerable<OrderDTO>> GetAllBusinessOrders() {
+        long? businessId = JwtTokenHandler.GetTokenBusinessId(HttpContext.Request.Headers.Authorization!);
+        if (businessId == null) return Forbid();
+
+        try {
+            var orderDTOs = _orderService.GetAllBusinessOrders((long)businessId);
+
+            return Ok(orderDTOs);
+        }
+        catch (Exception ex) {
+            _logger.LogError(ex, "An unexpected internal server error occured while getting all orders.");
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+        }
+    }
+
+    [HttpGet("active")]
+    [Authorize(Roles = "Admin,Owner,Employee")]
+    [ProducesResponseType<OrderDTO>(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public ActionResult<IEnumerable<OrderDTO>> GetAllActiveOrders() {
+        long? businessId = JwtTokenHandler.GetTokenBusinessId(HttpContext.Request.Headers.Authorization!);
+        if (businessId == null) return Forbid();
+
+        try {
+            var orderDTOs = _orderService.GetAllActiveOrders((long)businessId);
+
+            return Ok(orderDTOs);
+        }
+        catch (Exception ex) {
+            _logger.LogError(ex, "An unexpected internal server error occured while getting all active orders.");
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+        }
+    }
+
+    [HttpDelete("{id}")]
+    [Authorize(Roles = "Admin,Owner,Employee")]
+    [ProducesResponseType<OrderDTO>(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
```

Comment:
Are 404 and 422 response types even possible here? If not, please delete redundant attributes

Reply:
Yes, they are possible